retry/retry.go

```diff
@@ -48,6 +48,18 @@ func NewRetrier(name string,
 		notifyShouldNotRetryFuncs: []RetryNotifier{logShouldNotRetry}}
 }
 
+func New(name string,
+	backOffOpts *BackOffOpts, shouldRetryFunc func(error) bool) *Retrier {
+
+	return &Retrier{
+		Name:                      fmt.Sprintf("%s%d", name, atomic.AddUint32(&retrierNum, 1)),
```

Comment:
Minor, but I would move this into a `func newName` since it's now duplicated in some places.